NETOBSERV-2163 UI rework using tview

jpinsonneau opened the pull request:
## Description

- [x] switch rendering to tview
- [x] support background run
- [x] fix tests
- [x] fix e2e


![Screenshot From 2025-05-27 09-54-26](https://github.com/user-attachments/assets/0d8ae76d-5386-460c-9ffb-d09d7266c13e)
![Screenshot From 2025-05-27 09-55-01](https://github.com/user-attachments/assets/551a2cd0-4337-4ca1-881a-281fd1ff3bd6)
![Screenshot From 2025-05-27 09-55-34](https://github.com/user-attachments/assets/e20f4ac4-64e2-4faf-8bc0-825af582a9dd)
![Screenshot From 2025-05-27 11-50-29](https://github.com/user-attachments/assets/eda41bb4-ebc7-4703-b4ea-7a930a8ac2d5)


## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [X] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

jpinsonneau commented:
I finally found a way to fix the blinking effect :partying_face: 

The unit tests been simplified to avoid invoking tview but the e2e ones still need some rework here

jpinsonneau commented:
Rebased and addressed feedback from last demo

memodi commented:
@jpinsonneau - could you rebase please?

memodi commented:
@jpinsonneau [integration tests](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/netobserv_network-observability-cli/215/pull-ci-netobserv-network-observability-cli-main-integration-tests/1948042792227835904) are failing for the regexes, has something changed here?

jpinsonneau commented:
> @jpinsonneau [integration tests](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/netobserv_network-observability-cli/215/pull-ci-netobserv-network-observability-cli-main-integration-tests/1948042792227835904) are failing for the regexes, has something changed here?

I don't think so... I'm investigating on that :wink:

jpinsonneau commented:
/retest

jpinsonneau commented:
These tests are fine locally so I suspect the UI to crash because of the environment. I removed the exit when this happens and only display raw info in the console
https://github.com/netobserv/network-observability-cli/pull/215/commits/545339640d341891318547c5477649fa02f9a3b3

Let's see the test result with that new behavior

memodi commented:
it looks like [integration-tests passed ](https://prow.ci.openshift.org/pr-history/?org=netobserv&repo=network-observability-cli&pr=215) on your commit [5453396](https://github.com/netobserv/network-observability-cli/pull/215/commits/545339640d341891318547c5477649fa02f9a3b3)

jpinsonneau commented:
> > @memodi it passes when I keep the collector running on UI error.
> 
> you mean you have to keep running the collector in netobserv-cli for it to pass?
> 
> > Is there an easy way to get this error from the artifacts ?
> 
> are you looking to get collector pod logs? or logs from the test run?

Yeah the UI is crashing but now [I'm keeping the code running and only append logs](https://github.com/netobserv/network-observability-cli/pull/215/files#diff-a84f5928dfb1b5702387212c8bc2beafba4b77b1833ed5a9ee21797436e389fcR110-R114):
```
	errAdvancedDisplay = app.Run()
	if errAdvancedDisplay != nil {
		log.Errorf("Can't display advanced UI: %v", errAdvancedDisplay)
		app = nil
		hearbeat()
	}
```
I want to retreive the UI crash error to try to solve it in integration tests.

Collector pod logs or logs coming out from the cmd run call :wink:

jpinsonneau commented:
Reverted to exit on UI crash as discussed with @memodi 

Let me know when integration tests are all running even when one fail so we can confirm it's related to UI crash

memodi commented on e2e/integration-tests/integration_test.go:
```diff
@@ -5,19 +5,32 @@ import (
 	cliNS := "netobserv-cli"
 
+	g.BeforeAll(func() {
+		createErr := os.MkdirAll(path.Join("output"), 0700)
```
in CI, `$ARTIFACT_DIR` is exposed where each step artifacts can be stored, could you write to that dir?

jpinsonneau replied:
Sure !

jpinsonneau commented:
> ci-op-jwwcy07r

I don't have access to that :laughing:

jpinsonneau commented:
Rebased without changes

jpinsonneau commented:
Rebased and added timeouts to the tests since cmd.Run() wait for process to finish
https://github.com/netobserv/network-observability-cli/pull/215/commits/aa705d0cb7dd546e7f7ab8fd5237f78d6e318913

jpinsonneau commented:
@Amoghrd could you please have a look at integrations tests here ?

I have fixed those to support TTY to run the new UI and it works fine locally using kind.

Amoghrd commented:
I dont have much idea about those tests, can take a look over the week. If its the pods not coming up it might ImgPull issues for the controller or the collector pods
But if it was running previously and failing now, then something might have broken in this PR or something in konflux such that it cannot pull those images

Amoghrd commented:
/ok-to-test

Amoghrd commented:
/label qe-approved
